Decoupled stdlib from vm, script and compiler

earncef opened the pull request:
stdlib is fully optional. this is in prep for #68

d5 commented:
Hey, nice work as always. I have 2 comments:
1. I think we don't have to remove `stdlib` from tests. Test packages will not be included when Tengo is imported by other packages. 
2. What's your plan for `cmd/tengo`? We will have to ship 2 different binaries? One with stdlib, another without stdlib?

earncef commented:
1. I removed it from tests and moved some to stdlib (tests for os) not because of importing but for improved separation of concerns. The tests in vm should ideally be testing the vm itself and not the stdlib and the same goes for scripts. Stdlib now has some of the tests that were originally in the vm that test the stdlib functionality. I think stdlib might even do with some more tests but that’s a separate issue altogether.
2. Yes, IMO binaries would probably be the most straightforward way to deal with this.

earncef commented:
Also, I’ll try to work on improving the tests in the packages at some point. There’s some redundant tests that I think can be cleaned up and the focus narrowed a bit more tightly.

d5 commented:
Ok. Makes sense. We will have to update documentations, `cmd/tengo` _(likely just another package to include all stdlib?)_, and, also the build scripts accordingly.

d5 commented:
@earncef hey what's your email address? I'd like to invite you to Slack group.

earncef commented:
earncefs-at-yahoo

earncef commented:
Tengo binary is quite small. I'm not entirely sure the second binary without stdlib bundled is worth it. A flag in the main binary to disable stdlib might be enough.
